### ISSUE TITLE
Script editor: CTRL+Click on extends with relative path provide Errors

### ISSUE THREAD

**Issue by allkhor** [godotengine/godot#22224]
**Godot version:**
Godot 3.1 f148e8eed, platform=x11 bits=64 target=release_debug tools=yes

**OS/device including version:**
Xubuntu 16.04.5 LTS x64, 4.4.0-135-lowlatency

**Issue description:**
When ctrl+click on extends with relative path(extends "base.gd"), Godot generate errors:
```
ERROR: _load: No loader found for resource: res://b
   At: core/io/resource_loader.cpp:192.
ERROR: _load: No loader found for resource: res://ba
   At: core/io/resource_loader.cpp:192.
ERROR: _load: No loader found for resource: res://base
   At: core/io/resource_loader.cpp:192.
```

**Steps to reproduce:**
Open attached project and ctrl+click on "base.gd"

![click_on_extends](https://user-images.githubusercontent.com/1049442/45705106-87a69880-bb9a-11e8-98f9-458592805ee9.gif)

**Minimal reproduction project:**
[click_on_extends.zip](https://github.com/godotengine/godot/files/2393866/click_on_extends.zip)

**Comment by akien-mga** [comment]
Yes UTF-8 is allowed (and used by default). I'm not sure it's Unicode-related though, it looks like the ctrl+click code takes the word from the previous quote up to the point location, instead of going up to the next quote after the pointer.

**Comment by evandropoa** [comment]
Whell the parser fails to read the full constant after _extends_ in the example, stopping at the dot, if the _if_ statement at line **785:gdscript_tokenizercomment.cpp** is commented, it reads the full string:
`else if (CharType(GETCHAR(i)) == 0xFFFF) {
						//string ends here, next will be TK
						i--;
						break;
					}
`

But even after it has been parsed the correct way, it fails again, the correct resource loader cant be found becouse of the comparision between the ansi and unicode string, reporting they aren´t the same. line **69:resource_loader.cpp**:
`if (E->get().nocasecmp_to(extension) == 0)`

Sry, my english isnt very good, I hope you could understand,

**Comment by evandropoa** [comment]
Ok, I got it, the code 0xFFFF in the parser is the cursor inserted in the text. so inside a string I think it must be ignored, or postpone to the  end of string.

**Comment by evandropoa** [comment]
There´s someone i can talk about this issue and the solution i came for it?

 It´s related to the gdscript tokenizer, parser and code completion.

**Comment by KoBeWi** [comment]
Fixed in 3.2.3
